transformer_engine/pytorch/ops/fuser.py

```diff
@@ -132,9 +190,22 @@ def backward(
             ctx._saved_tensors_range = None
         del saved_tensors
 
+        # Unflatten list of extra tensor output grads
+        if len(grad_extra_outputs) != func_ctx.num_extra_outputs:
+            raise ValueError(
+                f"Expected grads for {func_ctx.num_extra_outputs} extra tensor outputs, "
+                f"but got {len(grad_extra_outputs)}"
+            )
+        basic_op_grad_extra_outputs = []
+        for op in basic_ops:
+            dys, grad_extra_outputs = _split_tuple(grad_extra_outputs, op.num_extra_outputs)
+            basic_op_grad_extra_outputs.append(dys)
+            del dys
```

Comment:
Why?

Reply:
Overoptimization. My thought was to aggressively dereference these tensors so that PyTorch can deallocate these buffers. However, it wouldn't have an effect since we don't dereference tensors in `basic_op_grad_extra_outputs`.